### PR TITLE
warehouse: Use semantic forms for webauthn actions

### LINKED ISSUE #6472: form on webauthn page is not a <form>
On `warehouse/templates/manage/account/webauthn-provision.html` we have a form where users can provision their device.

This is *not* wrapped in `<form>` tags, which means it is not semantic.

### PR DESCRIPTION
**Pull request by woodruffw**
cc @di @nlhkabu 

Fixes #6472.

### CODE REVIEW

**Comment by nlhkabu** [comment]
Thank you @woodruffw.  Ok on my side, but I will wait for @di to review for the JS :)

File: warehouse/static/js/warehouse/utils/webauthn.js [hunk 1]
```diff
@@ -30,18 +30,23 @@ const populateWebAuthnErrorList = (errors) => {
...
+  if (webAuthnForm === null) {
     return null;
   }
 
+  const webAuthnButton = webAuthnForm.querySelector("button[type=submit]");
   webAuthnButton.disabled = false;
-  webAuthnButton.addEventListener("click", async () => { func(webAuthnButton.value); });
+
+  webAuthnForm.addEventListener("submit", async() => {
+    func(webAuthnButton.value);
```
Comment (webknjaz) [inline warehouse/static/js/warehouse/utils/webauthn.js]: This should probably be:
```suggestion
    await func(webAuthnButton.value);
```

Comment (nlhkabu) [reply warehouse/static/js/warehouse/utils/webauthn.js]: Thanks @webknjaz.
If you want to make a new PR with this update, please do so :)

Comment (webknjaz) [reply warehouse/static/js/warehouse/utils/webauthn.js]: Here you go: https://github.com/pypa/warehouse/pull/6608

Comment (nlhkabu) [reply warehouse/static/js/warehouse/utils/webauthn.js]: thanks!

**Pull request merged**